### PR TITLE
Config refactor

### PR DESCRIPTION
**Pull request by ajvb**
This refactors the separate exemptions and severity config into a single big YAML config. It also adds the ability to mark regressions and have security group specific whitelisted ports (fixing #92).

(I have not finished the changes required to the `README.md`. Would prefer to wait for the initial review before finishing it up.)

r? @g-k - mind giving this an initial once over?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -176,27 +175,70 @@ head .cache/v/pytest_aws:cloudservices-aws-stage:us-west-2:rds:describe_db_insta
...
+  whitelisted_ports_global:
+    - 25
+  whitelisted_ports:
+    - test_param_id: '*bastion'
+      ports:
+        - 22
+        - 2222
+```
+
+The `--aws-require-tag` and `--aws-whitelisted-ports` will overwrite whatever is present in the config file.
```
Comment (g-k) [inline README.md]: We could just nuke these options too if you want to put everything in the config file.

**Comment by ajvb** [comment]
@g-k thanks for the initial review. Also made the suggested change and just removed the CLI options. Ok I'd say this is ready for CR.

File: README.md [hunk 2]
```diff
@@ -62,11 +62,8 @@ and options pytest-services adds:
 * `--aws-debug-calls` for printing (with `-s`) API calls we make
 * `--aws-profiles` for selecting one or more AWS profiles to fetch resources for or the AWS default profile / `AWS_PROFILE` environment variable
 * `--aws-regions` for selecting one or more AWS regions to fetch resources from or the default of all regions
-* `--aws-require-tag` adds Tag names to check on all EC2 instances for the `aws.ec2.test_ec2_instance_has_required_tags` test
-* `--aws-whitelisted-ports` adds ports to the whitelist for the `aws.ec2.test_ec2_security_group_opens_specific_ports_to_all` test
 * `--offline` a flag to tell HTTP clients to not make requests and return empty params
-* [`--severity-config`](#test-severity) path to a config file for marking different tests with different severities
-* [`--exemptions-config`](#test-exemptions) path to a config file for marking test name and resource IDs as expected failures
+* [`--config`](#custom-test-config) path to test custom config file
```
Comment (g-k) [inline README.md]: nit: might be handy to link to the "Custom Test Config" from here

Comment (ajvb) [reply README.md]: as in the `config.yaml.example` file? Or the section in the README?

File: conftest.py [hunk 3]
```diff
@@ -33,42 +30,24 @@ def pytest_addoption(parser):
...
-                     nargs='*',
-                     default=[],
-                     help='EC2 instance tags for the aws.ec2.test_ec2_instance_has_required_tags test to check.')
-
-    parser.addoption('--aws-whitelisted-ports',
-                     nargs='*',
-                     default=[],
-                     help='Additional ports to whitelist for '
-                          'aws.ec2.test_ec2_security_group_opens_specific_ports_to_all test.')
-
```
Comment (g-k) [inline conftest.py]: 😄

File: regressions.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    ... 'comment': 'foo bar baz!'
+    ... },
+    ... {
+    ... 'test_name': 'test_foo',
+    ... 'test_param_id': 'foo-id',
+    ... 'comment': 'bar bar baz!'
+    ... }
+    ... ]) == {'test_foo': {'foo-id': 'foo bar baz!'}}
+    True
+    >>> # UserWarning: Regressions: test_name: test_foo | test_id: foo-id | Skipping duplicate test name and ID
```
Comment (g-k) [inline regressions.py]: 👍

File: README.md (hunk 2 above)
Comment (g-k) [reply README.md]: I was thinking that that section of the readme

Comment (ajvb) [reply README.md]: Oh ok, yeah its doing that with the `(#custom-test-config)` bit, like it was with the `(#test-severity)` bit before. Do you mean having the link text be different? Or is what's there alright?

Comment (g-k) [reply README.md]: d'oh missed that nm, what's there is good.

**Pull request merged**